### PR TITLE
Add Barbar support

### PR DESCRIPTION
**Pull request by James-DeLorenzo**
sample [barbar](https://github.com/romgrk/barbar.nvim) bufferline;
sample_barbar_9

- plugins.lua is an inactive buffer, not shown in any current window
- test.py was the last buffer visible, using the alternate highlighting (when highlight_alertnate is false in settings this highlight does not show and shows like plugins.lua)
- init.lua is the currently open buffer
- bluloco-dark.lua is a buffer that is visible in a window but not the active window/buffer

samples via `Lushify` call;
sample_barbar_6
sample_barbar_7

### CODE REVIEW

File: lua/lush_theme/bluloco.lua [hunk 1]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
     -- BufferLineHintSelected { fg = t.hint, sp = t.primary, gui = "underline" },
 
+    -- BarBar
+
+    BufferCurrent { fg = t.fg, bg = t.bg },
+    BufferCurrentIndex { BufferCurrent },
+    BufferCurrentIcon { BufferCurrentIndex },
+    BufferCurrentMod { BufferCurrent },
+    BufferCurrentSign { fg = t.keyword, bg = BufferCurrent.bg },
+    BufferCurrentTarget { fg = shade(t.error, 40), bg = BufferCurrent.bg },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]: Let's use the red from types, and stay consistent for all buffers so it's easier for the eyes? I have added suggestsions for those down below also.
```suggestion
    BufferCurrentTarget { BufferCurrent, fg = t.type },
```

File: lua/lush_theme/bluloco.lua [hunk 2]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferCurrentINFO { fg = DiagnosticInfo.fg, bg = BufferCurrent.bg },
+    BufferCurrentERROR { fg = DiagnosticError.fg, bg = BufferCurrent.bg },
+    BufferCurrentHINT { fg = DiagnosticHint.fg, bg = BufferCurrent.bg },
+
+    BufferInactive { fg = t.shade40, bg = t.bgFloat },
+    BufferInactiveIcon { BufferInactive },
+    BufferInactiveIndex { BufferInactive },
+    BufferInactiveMod { BufferInactive },
+    BufferInactiveSign { BufferInactive },
+    BufferInactiveTarget { fg = shade(BufferInactive.fg, 70), bg = BufferInactive.bg },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]:
```suggestion
    BufferInactiveTarget { BufferInactive, fg = t.type },
```

File: lua/lush_theme/bluloco.lua [hunk 3]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferInactiveINFO { BufferCurrentINFO, bg = BufferInactive.bg },
+    BufferInactiveERROR { BufferCurrentERROR, bg = BufferInactive.bg },
+    BufferInactiveHINT { BufferCurrentHINT, bg = BufferInactive.bg },
+
+    BufferVisible { BufferCurrent, bg = t.bgFloat },
+    BufferVisibleIndex { BufferVisible },
+    BufferVisibleIcon { BufferVisibleIndex },
+    BufferVisibleMod { BufferVisible },
+    BufferVisibleSign { BufferVisible },
+    BufferVisibleTarget { fg = shade(BufferVisible.fg, 70), bg = BufferVisible.bg },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]:
```suggestion
    BufferVisibleTarget { BufferVisible, fg = t.type },
```

File: lua/lush_theme/bluloco.lua [hunk 4]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferVisibleINFO { BufferInactiveINFO },
+    BufferVisibleERROR { BufferInactiveERROR },
+    BufferVisibleHINT { BufferInactiveHINT },
+
+    BufferAlternate { BufferInactive, fg = t.tag },
+    BufferAlternateIndex { BufferAlternate },
+    BufferAlternateIcon { BufferAlternateIndex },
+    BufferAlternateMod { BufferInactiveMod },
+    BufferAlternateSign { BufferInactiveSign },
+    BufferAlternateTarget { BufferInactiveTarget, fg = shade(BufferAlternate.fg, 70) },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]:
```suggestion
    BufferAlternateTarget { BufferAlternate, fg = t.type},
```

File: lua/lush_theme/bluloco.lua [hunk 5]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferVisibleIcon { BufferVisibleIndex },
+    BufferVisibleMod { BufferVisible },
+    BufferVisibleSign { BufferVisible },
+    BufferVisibleTarget { fg = shade(BufferVisible.fg, 70), bg = BufferVisible.bg },
+    BufferVisibleWARN { BufferInactiveWARN },
+    BufferVisibleINFO { BufferInactiveINFO },
+    BufferVisibleERROR { BufferInactiveERROR },
+    BufferVisibleHINT { BufferInactiveHINT },
+
+    BufferAlternate { BufferInactive, fg = t.tag },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]: I think this is just too much for a alternate buffer, probalby the sign only would be better, see below
```suggestion
    BufferAlternate { BufferInactive },
```

File: lua/lush_theme/bluloco.lua [hunk 6]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferAlternateIndex { BufferAlternate },
+    BufferAlternateIcon { BufferAlternateIndex },
+    BufferAlternateMod { BufferInactiveMod },
+    BufferAlternateSign { BufferInactiveSign },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]: blue would be just too close to the actual current one. how about purple for alternate sign?
```suggestion
    BufferAlternateSign { BufferInactiveSign, fg = t.constant },
```

File: lua/lush_theme/bluloco.lua [hunk 7]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
     -- BufferLineInfoSelected { fg = t.info, sp = t.primary, gui = "underline" },
     -- BufferLineHintSelected { fg = t.hint, sp = t.primary, gui = "underline" },
 
+    -- BarBar
+
+    BufferCurrent { fg = t.fg, bg = t.bg },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]: I'd like the bufferline to be transparent also if enabled, so using Normal here would be better
```suggestion
    BufferCurrent { Normal },
```

File: lua/lush_theme/bluloco.lua [hunk 8]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferAlternateMod { BufferInactiveMod },
+    BufferAlternateSign { BufferInactiveSign },
+    BufferAlternateTarget { BufferInactiveTarget, fg = shade(BufferAlternate.fg, 70) },
+    BufferAlternateWARN { BufferInactiveWARN },
+    BufferAlternateINFO { BufferInactiveINFO },
+    BufferAlternateERROR { BufferInactiveERROR },
+    BufferAlternateHINT { BufferInactiveHINT },
+
+    BufferTabpages { BufferInactive, fg = t.shade90 },
+    BufferTabpageFill { BufferTabpages, fg = t.shade3 },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]:
```suggestion
    BufferTabpageFill { BufferTabpages, fg = t.bg },
```

File: lua/lush_theme/bluloco.lua [hunk 9]
```diff
@@ -422,6 +422,55 @@ local theme = lush(function(injected_functions)
...
+    BufferAlternateIcon { BufferAlternateIndex },
+    BufferAlternateMod { BufferInactiveMod },
+    BufferAlternateSign { BufferInactiveSign },
+    BufferAlternateTarget { BufferInactiveTarget, fg = shade(BufferAlternate.fg, 70) },
+    BufferAlternateWARN { BufferInactiveWARN },
+    BufferAlternateINFO { BufferInactiveINFO },
+    BufferAlternateERROR { BufferInactiveERROR },
+    BufferAlternateHINT { BufferInactiveHINT },
+
+    BufferTabpages { BufferInactive, fg = t.shade90 },
```
Comment (uloco) [inline lua/lush_theme/bluloco.lua]:
```suggestion
    BufferTabpages { BufferInactive, fg = t.fg },
```

**Review by uloco — CHANGES_REQUESTED** [review]
Thank you so much for this PR :)

I have some minor nitpicks, that would improve things in my opinion:

* You could use t.type als a fg for all target highlights, so it's better visible
* I think the alternate buffer is too visible, how about we keep the text as BufferInactive but just highlight the BufferAlternateSign with t.constant?
* To make this also work with transparent mode, we should add the bgs to the init.lua transparent section.
```lua
BufferInactive { theme.BufferInactive, bg = "NONE" },
BufferVisible { BufferCurrent }
```

File: lua/lush_theme/bluloco.lua (hunk 6 above)
Comment (James-DeLorenzo) [reply lua/lush_theme/bluloco.lua]: ok, here's how that looks now;
image

Comment (uloco) [reply lua/lush_theme/bluloco.lua]: I like it, thanks :)

**Review by uloco — APPROVED** [review]
Love it, thank you so much again :)

**Pull request merged**